src/tests/ftest/nvme/enospace.py

```diff
@@ -22,52 +24,68 @@ class NvmeEnospace(ServerFillUp):
+        self.test_result = []
         self.daos_cmd = None
 
     def setUp(self):
-        """Initial setup"""
+        """Initial setup."""
         super().setUp()
 
+        self.test_result = []
+
         # initialize daos command
         self.daos_cmd = DaosCommand(self.bin)
         self.create_pool_max_size()
-        self.der_nospace_count = 0
-        self.other_errors_count = 0
-        self.test_result = []
 
-    def verify_enospace_log(self, der_nospace_err_count):
-        """
-        Function to verify there are no other error except DER_NOSPACE and
-        DER_NO_HDL in client log.Verify DER_NOSPACE count is higher.
+    def verify_enospace_log(self, log_file):
+        """Function checking logs consistency.
+
+        Function checking that only expected errors have occurred and the DER_NOSPACE errors have
+        occurred.
 
-        args:
-            expected_err_count(int): Expected DER_NOSPACE count from client log.
+        Args:
+            log_file (str): name prefix of the log files to check.
         """
-        # Get the DER_NOSPACE and other error count from log
-        self.der_nospace_count, self.other_errors_count = error_count(
-            "-1007", self.hostlist_clients, self.client_log)
-
-        # Get the DER_NO_HDL and other error count from log
-        der_nohdl_count, other_nohdl_err = error_count(
-            "-1002", self.hostlist_clients, self.client_log)
-
-        # Check there are no other errors in log file except DER_NO_HDL
-        if self.other_errors_count != der_nohdl_count:
-            self.fail('Found other errors, count {} in client log {}'
-                      .format(int(self.other_errors_count - other_nohdl_err),
-                              self.client_log))
-        # Check the DER_NOSPACE error count is higher if not test will FAIL
-        if self.der_nospace_count < der_nospace_err_count:
-            self.fail('Expected DER_NOSPACE should be > {} and Found {}'
-                      .format(der_nospace_err_count, self.der_nospace_count))
+        logfile_glob = log_file + r".*[0-9]"
+        errors_count = get_errors_count(self.hostlist_clients, logfile_glob)
+        for error in self.expected_errors:
+            if error not in errors_count:
+                errors_count[error] = 0
+
+        if errors_count[self.DER_NOSPACE] <= 0:
+            self.fail(
+                "Expected DER_NOSPACE (-1007) should be > 0: got={}"
+                .format(errors_count[self.DER_NOSPACE]))
+
+        if len(self.expected_errors) != len(errors_count):
+            unexpected_errors_count = {
+                key: val for key, val in errors_count.items() if key not in self.expected_errors
+            }
+            msg = 'Found unexpected errors in client logs {}: '.format(logfile_glob)
+            msg += ", ".join(
+                f'{c_err_to_str(key)}({key}): got={val}'
+                for key, val in unexpected_errors_count.items())
```

Comment:
If you know what was expected and you know what you got then why do you need to call `c_err_to_str`?

Reply:
The unexpected errors will eventually contains unexpected errors which will fail the test.
In such situation, it will be easier to debug it if we have the DAOS name of the error and not just its error code.